frontend/src/metabase-lib/display.ts

```diff
@@ -0,0 +1,168 @@
+
+  if (isNative) {
+    return { display: "table" };
+  }
+
+  const stageIndex = -1;
+  const aggregations = Lib.aggregations(query, stageIndex);
+  const breakouts = Lib.breakouts(query, stageIndex);
+
+  if (aggregations.length === 0 && breakouts.length === 0) {
+    return { display: "table" };
+  }
+
+  if (aggregations.length === 1 && breakouts.length === 0) {
+    return { display: "scalar" };
+  }
+
+  if (aggregations.length === 1 && breakouts.length === 1) {
+    const [{ columnInfo }] = getBreakoutsWithColumnInfo(
+      query,
+      stageIndex,
+      breakouts,
+    );
+
+    if (isState(columnInfo)) {
+      return {
+        display: "map",
+        settings: {
+          "map.type": "region",
+          "map.region": "us_states",
+        },
+      };
+    }
+
+    if (isCountry(columnInfo)) {
+      return {
+        display: "map",
+        settings: {
+          "map.type": "region",
+          "map.region": "world_countries",
+        },
+      };
+    }
+  }
+
+  if (aggregations.length >= 1 && breakouts.length === 1) {
+    const [{ breakout, columnInfo }] = getBreakoutsWithColumnInfo(
+      query,
+      stageIndex,
+      breakouts,
+    );
+
+    if (isDate(columnInfo)) {
+      const info = Lib.displayInfo(query, stageIndex, breakout);
+
+      if (info.isTemporalExtraction) {
+        return { display: "bar" };
+      }
+
+      return { display: "line" };
+    }
+
+    const binning = Lib.binning(breakout);
+    const isBinned = binning !== null;
+
+    if (isBinned) {
+      return { display: "bar" };
+    }
+
+    if (isCategory(columnInfo)) {
+      return { display: "bar" };
+    }
+  }
+
+  if (aggregations.length === 1 && breakouts.length === 2) {
+    const breakoutsWithColumnInfo = getBreakoutsWithColumnInfo(
+      query,
+      stageIndex,
+      breakouts,
+    );
+
+    const isAnyBreakoutDate = breakoutsWithColumnInfo.some(({ columnInfo }) => {
+      return isDate(columnInfo);
+    });
+    if (isAnyBreakoutDate) {
+      return { display: "line" };
+    }
+
+    const areBreakoutsCoordinates = breakoutsWithColumnInfo.every(
+      ({ columnInfo }) => {
+        return isCoordinate(columnInfo);
+      },
+    );
+    if (areBreakoutsCoordinates) {
+      return {
+        display: "map",
+        settings: {
+          "map.type": "grid",
+        },
+      };
+    }
+
+    const areBreakoutsCategories = breakoutsWithColumnInfo.every(
+      ({ columnInfo }) => {
+        return isCategory(columnInfo);
+      },
+    );
+    if (areBreakoutsCategories) {
+      return { display: "bar" };
+    }
+  }
+
+  return { display: "table" };
+};
+
+const getBreakoutsWithColumnInfo = (
+  query: Lib.Query,
+  stageIndex: number,
+  breakouts: Lib.BreakoutClause[],
+) => {
+  return breakouts.map(breakout => {
+    const column = Lib.breakoutColumn(query, stageIndex, breakout);
+    const columnInfo = Lib.displayInfo(query, stageIndex, column);
+    return { breakout, columnInfo };
```

Comment:
```suggestion
    return { breakout, column };
```

based on discussion in [slack](https://metaboat.slack.com/archives/C0645JP1W81/p1706864885372559?thread_ts=1706862717.614449&cid=C0645JP1W81), I believe we don't need to call displayInfo, but return a column itself instead, then check type of the column with methods available in `metabase-lib/column_types`

Reply:
7df3d78d24efba26a1a6f658572dad59ae299173